src/orders/components/OrderHistory/OrderHistory.tsx

```diff
@@ -128,39 +197,167 @@ const OrderHistory = ({
                       onNoteUpdateLoading={onNoteUpdateLoading!}
                       relatedId={related.id}
                       id={id}
-                      date={date}
-                      user={user}
+                      date={<OrderHistoryDate date={date} />}
+                      actor={toActor(user, app)}
                       message={message}
                       key={id}
-                      app={app}
+                      eventData={event}
+                      isLastInGroup={isLastInGroup}
                     />
                   );
                 }
 
+                if (
+                  type === OrderEventsEnum.ORDER_MARKED_AS_PAID ||
+                  type === OrderEventsEnum.ORDER_FULLY_PAID
+                ) {
+                  const hasTransactionRef = !!event.transactionReference;
+                  const hasContent = hasTransactionRef || message;
+
+                  return (
+                    <TimelineEvent
+                      date={<OrderHistoryDate date={date} />}
+                      eventData={event}
+                      actor={toActor(user, app)}
+                      eventType={type}
+                      isLastInGroup={isLastInGroup}
+                      title={
+                        <Box display="flex" gap={1} alignItems="center" flexWrap="wrap">
+                          <FormattedMessage id="KpP/aW" defaultMessage="Order was marked as" />
```

Comment:
suggestion: This can get tricky for translations, if for some reason "paid" would need to be before this string for a proper sentence.

We can use react-intl for this as well, example:

```ts
<FormattedMessage
    id="order.statusChange"
    defaultMessage="Order was marked as {status}"
    values={{
      // We define 'status' as the actual Pill component
      status: (
        <Pill
          color="success"
          size="small"
          // The label inside the pill is still translated normally
          label={intl.formatMessage({
            id: "u/vOPu",
            defaultMessage: "Paid",
          })}
        />
      ),
    }}
  />
```